### ISSUE TITLE
Second display framebuffer created when using virtio-ramfb-gl

### ISSUE THREAD

**Issue by phosphorcube** [utmapp/UTM#4438]
I've encountered this issue when using Fedora with KDE Plasma, however it might affect other distros as well. A second display framebuffer is created and shares the same VM window, which is very annoying to disable because it causes the mouse movement to be off.

**Configuration**  
* UTM Version: 3.2.4
* macOS Version: 12.6
<img width="1392" alt="Screen Shot 2022-09-22 at 10 05 42 PM" src="https://user-images.githubusercontent.com/18562602/191895566-66b0de43-2149-4302-b1ea-b4ebcfe3b1e0.png">

* Mac Chip (Intel, M1, ...): M1
[debug log and config plist.zip](https://github.com/utmapp/UTM/files/9630839/debug.log.and.config.plist.zip)

**Comment by phosphorcube** [comment]
PS: For anyone looking for a work around, you can disable that second display, but it's annoying and difficult to do because the mouse movement is off, however once you've got it disabled then the mouse movement will be fixed.

**Comment by romi2002** [comment]
Same issue here, on Fedora GNOME and UTM `4.1.0`.

**Comment by romi2002** [comment]
A temporary (but hacky) fix, is to place a `.desktop` file with the following in `/etc/xdg/autostart`. This is so the second display is disabled on gnome login.
```
[Desktop Entry]
Type=Application
Name=Xrandr
Exec=xrandr --output None-1-1 --off
Terminal=false
```

**Comment by AwlsomeAlex** [comment]
Ditto. Fedora 37 with UTM 4.1.0.

**Comment by gasinvein** [comment]
It seems like the second output is the kernel's [simple framebuffer](https://www.kernel.org/doc/Documentation/devicetree/bindings/display/simple-framebuffer.txt), which was [enabled](https://fedoraproject.org/wiki/Changes/ReplaceFbdevDrivers) in Fedora 36.
As a workaround, you can disable it by adding `initcall_blacklist=sysfb_init` to the kernel boot arguments (press `e` while in GRUB boot menu, and type in the argument at the end of the line starting with `linux`, then press `Ctrl+x`).

**Comment by AwlsomeAlex** [comment]
This fixes it for me.

**Comment by gasinvein** [comment]
> Anaconda seems to crash now

Yeah, I didn't find a way around it other than switching the gpu device to `virtio-ramfb` before installing Fedora and then switching it back to `virtio-ramfb-gl`, or running `liveinst` from serial console and installing in text mode.

**Comment by osy** [comment]
Have you all tried the latest beta? Crashes due to graphics rendering should be fixed.

**Comment by osy** [comment]
“ renders unusably slow” explain?

**Comment by gasinvein** [comment]
> “ renders unusably slow” explain?

The display draws the image line-by-line, something like a really slow CRT. Even GRUB menu is like this.

**Comment by osy** [comment]
That’s always been an issue with UEFI firmware’s virtiogpu drivers. I’ll try to fix it one day but it shouldn’t affect anything else

> Anaconda still crashes

Do you mean crashes UTM or crashes in the guest? If it’s the guest try running it with QEMU. If it’s an issue with QEMU, file an issue with them. Otherwise open a new issue here.

**Comment by gasinvein** [comment]
> That’s always been an issue with UEFI firmware’s virtiogpu drivers.

Just curious, why doesn't it happen with `virtio-ramfb-gl` device?

> > Anaconda still crashes
> 
> Do you mean crashes UTM or crashes in the guest?

Crashes in the guest. And indeed it looks like an unrelated problem.

**Comment by osy** [comment]
> Just curious, why doesn't it happen with virtio-ramfb-gl device?

EFI uses the ramfb device instead of the virtiogpu device when both are provided.

**Comment by osy** [comment]
So I don’t have grub set to show up every boot so the slow console buffer isn’t an issue for me. If people really feel that strongly about it, I can investigate the issue in the EFI driver code…

**Comment by AwlsomeAlex** [comment]
I find it becomes more of an issue if you're using Retina mode and it has to draw the entire GRUB menu. If this GPU driver is the new default, it should probably be documented in "Known Issues", as the workaround isn't apparent.

**Comment by gasinvein** [comment]
I can confirm it seems to work properly on 4.2.4. Tested with Fedora 38, GNOME Wayland session, booted with `initcall_blacklist=sysfb_init`; used `glmark2` and `glmark2-wayland` programs.
Oddly enough and contrary to the known issues list, it seems to work fine with the Metal ANGLE backend as well.
